e2e/e2e_test.go

```diff
@@ -106,9 +106,97 @@ var _ = Describe("Whereabouts functionality", func() {
 
 			It("allocates a single pod within the correct IP range", func() {
 				By("checking pod IP is within whereabouts IPAM range")
-				secondaryIfaceIP, err := retrievers.SecondaryIfaceIPValue(pod)
+				secondaryIfaceIPs, err := retrievers.SecondaryIfaceIPValue(pod)
 				Expect(err).NotTo(HaveOccurred())
-				Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(Succeed())
+				Expect(inRange(ipv4TestRange, secondaryIfaceIPs[0])).To(Succeed())
+			})
+		})
+
+		Context("DualStack", func() {
+			const (
+				testDualStackNetworkName = "wa-dualstack-nad"
+				dualStackIPv4Range       = "11.11.0.0/16"
+				dualStackIPv6Range       = "abcd::0/64"
+			)
+
+			var (
+				netAttachDefDualStack *nettypes.NetworkAttachmentDefinition
+				testIPRangesDualStack = []string{dualStackIPv4Range, dualStackIPv6Range}
+			)
+
+			Context("IPRanges configuration only", func() {
+				BeforeEach(func() {
+					const dualstackPodName = "whereabouts-dualstack-test"
+					var err error
+
+					netAttachDefDualStack = macvlanNetworkWithWhereaboutsIPAMNetwork(testDualStackNetworkName, testNamespace, "", testIPRangesDualStack)
+
+					By("creating DualStack NetworkAttachmentDefinition for whereabouts")
+					_, err = clientInfo.AddNetAttachDef(netAttachDefDualStack)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("creating a pod with whereabouts net-attach-def")
+					pod, err = clientInfo.ProvisionPod(
+						dualstackPodName,
+						testNamespace,
+						podTierLabel(dualstackPodName),
+						entities.PodNetworkSelectionElements(testDualStackNetworkName),
+					)
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					By("deleting pod with whereabouts net-attach-def")
+					Expect(clientInfo.DeletePod(pod)).To(Succeed())
+					By("deleting DualStack NetworkAttachmentDefinition for whereabouts")
+					Expect(clientInfo.DelNetAttachDef(netAttachDefDualStack)).To(Succeed())
+				})
+
+				It("allocates a single pod within the correct IP ranges", func() {
+					By("checking pod IP is within whereabouts IPAM ranges")
+					secondaryIfaceIPs, err := retrievers.SecondaryIfaceIPValue(pod)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(inRange(dualStackIPv4Range, secondaryIfaceIPs[0])).To(Succeed())
+					Expect(inRange(dualStackIPv6Range, secondaryIfaceIPs[1])).To(Succeed())
```

Comment:
Please assert the length of the slice before assuming the indexes: `Expect(secondaryIfaceIPs).To(HaveLen(2))`